### PR TITLE
Implement Range#last in Ruby

### LINKED ISSUE #208: Implement Range#last
https://ruby-doc.org/core-2.6.3/Range.html#method-i-last

Not sure if we can do this in Ruby.

The mruby implementation might be a good starting point.

https://github.com/artichoke/artichoke/blob/51f393863b93405a2ae2d4fdb34b80a2eb95055f/mruby-sys/vendor/mruby/mrbgems/mruby-range-ext/src/range.c#L84-L96

### PR DESCRIPTION
**Pull request by tkbky**
Fixes #208

### CODE REVIEW

File: artichoke-backend/src/extn/core/range/range.rb [hunk 1]
```diff
@@ -19,6 +19,21 @@ def first(*args)
     ary
   end
 
+  def last(*args)
+    return self.end if args.empty?
+
+    raise ArgumentError, "wrong number of arguments (given #{args.length}, expected 1)" unless args.length == 1
+
+    arg = args[0]
+    raise TypeError, "no implicit conversion of #{arg.class} into Integer" unless arg.respond_to?(:to_int)
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/range/range.rb]: this raise does the same thing with immediate objects like `nil` where it prints the value of `#inspect1 directly.

```
[2.6.3] > (0...1).last(nil)
Traceback (most recent call last):
        5: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /usr/local/var/rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):4
        1: from (irb):4:in `last'
TypeError (no implicit conversion from nil to integer)
```

File: artichoke-backend/src/extn/core/range/range.rb [hunk 2]
```diff
@@ -19,6 +19,21 @@ def first(*args)
     ary
   end
 
+  def last(*args)
+    return self.end if args.empty?
+
+    raise ArgumentError, "wrong number of arguments (given #{args.length}, expected 1)" unless args.length == 1
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/range/range.rb]: this should be `expected (0..1)`

```
[2.6.3] > (0...1).last(nil, nil)
Traceback (most recent call last):
        6: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        5: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `load'
        4: from /usr/local/var/rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        3: from (irb):5
        2: from (irb):5:in `rescue in irb_binding'
        1: from (irb):5:in `last'
ArgumentError (wrong number of arguments (given 2, expected 0..1))
```

File: artichoke-backend/src/extn/core/range/range.rb (hunk 1 above)
Comment (lopopolo) [reply artichoke-backend/src/extn/core/range/range.rb]: Also for booleans

Comment (tkbky) [reply artichoke-backend/src/extn/core/range/range.rb]: whoops, totally miss that. good call out. thanks.

File: artichoke-backend/src/extn/core/range/range.rb [hunk 3]
```diff
@@ -22,10 +22,12 @@ def first(*args)
   def last(*args)
     return self.end if args.empty?
 
-    raise ArgumentError, "wrong number of arguments (given #{args.length}, expected 1)" unless args.length == 1
+    raise ArgumentError, "wrong number of arguments (given #{args.length}, expected #{inspect})" unless args.length == 1
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/range/range.rb]: `inspect` prints debug information for the range object. I think we want the literal string `(0..1)`.

**Comment by tkbky** [comment]
It just my silly mistake. 😄

**Pull request merged**